chrome/content/zotero/bibliography.js

```diff
@@ -206,9 +224,43 @@ var Zotero_File_Interface_Bibliography = new function() {
 		window.sizeToContent();
 	}
 
+	/*
+	 * Update locale menulist when style is changed
+	 */
+	function updateLocaleMenu(selectedStyle) {
+		// For styles with a default-locale, disable locale menulist and show locale
+		var menulist = document.getElementById("locale-menu");
+		
+		// If not null, then menulist is extended with the default-locale value
+		// of the previously selected style
+		if (defaultStyleLocale) {
+			// Reset menulist
+			menulist.removeItemAt(0);
+			defaultStyleLocale = "";
+		}
+		
+		if (selectedStyle.locale) {
+			defaultStyleLocale = selectedStyle.locale;
+			
+			//add default-locale to menulist
+			var localeLabel = defaultStyleLocale;
+			if (Zotero.Styles.locales.hasOwnProperty(defaultStyleLocale)) {
```

Comment:
No real need to use ``hasOwnProperty`` unless you really need to avoid checking the prototype chain. (We also don't worry about an extended ``Object``.) Cleaner and easier (and, counterintuitively, [much faster](http://jsperf.com/hasownproperty-vs-in-vs-undefined/17), though it doesn't matter here) to just do ``if (Zotero.Styles.locales[defaultStyleLocale]) {``, or ``if (Zotero.Styles.locales[defaultStyleLocale] !== undefined) {`` in cases where truthy isn't sufficient.